File: answers/Python/@oseme-techguy/08-combine-two-files.py
"""
    Solution to Combine two files
"""

if __name__ == '__main__':
    fist_path = './data/08-combine-two-files-1.txt'
    second_path = './data/08-combine-two-files-2.txt'
    with open(fist_path, encoding='utf-8') as file1, open(second_path, encoding='utf-8') as file2:
        for line1, line2 in zip(file1, file2):
            print(line1.strip('\n') + ' ' + line2.strip('\n'))

File: answers/Go/@codehakase/08-combine-two-files.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"strings"
)

func main() {
	firstFile, err := ioutil.ReadFile("./data/08-combine-two-files-1.txt")
	if err != nil {
		log.Fatalf("failed to open file1: %v", err)
	}
	secondFile, err := ioutil.ReadFile("./data/08-combine-two-files-2.txt")
	if err != nil {
		log.Fatalf("failed to open file2: %v", err)
	}
	file1Words := strings.Split(string(firstFile), "\n")
	file2Words := strings.Split(string(secondFile), "\n")
	words := []string{}
	for i := 0; i < len(file1Words); i++ {
		word := fmt.Sprintf("%s %s\n", file1Words[i], file2Words[i])
		words = append(words, word)
	}
	fmt.Println(strings.Join(words, ""))
}
